File: Function.GraphQL/Shared/movie-api/dtos/provider.ts
export type Provider = {
  logo_path: string,
  provider_name: string,
  provider_id: number,
  display_priority: number
};

export type ProviderDisplay = Provider & {
  display_priorities: { [country: string]: number }
};

export type WatchProviders = {
  link: string,
  buy?: Provider[],
  rent?: Provider[],
  flatrate?: Provider[],
};

File: WatchWhat.Apple/Shared/Models/Provider.swift
//
//  WatchProvider.swift
//  WatchWhat
//
//  Created by Drew Miller on 3/11/23.
//

import Foundation

struct Providers {
    let link: String
    let buy: [Provider]?
    let rent: [Provider]?
    let flatrate: [Provider]?
}

struct Provider: Hashable {
    let id: Int
    let name: String
    let displayPriority: Int
    let logo: String?
}
